src/content/reference/react/components.md

```diff
@@ -22,3 +22,23 @@ React exposes a few built-in components that you can use in your JSX.
 ## Your own components {/*your-own-components*/}
 
 You can also [define your own components](/learn/your-first-component) as JavaScript functions.
+
+--
+
+## Client and Server Components {/*client-and-server-components*/}
+
+<Wip>
+[React Server Components](/learn/start-a-new-react-project#bleeding-edge-react-frameworks) are only available in React’s canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+</Wip>
+
+There are two main types of components in React: Server Components and Client Components. Both types follow the same basic principles of React - they are reusable, composable, and stateful - but they are used in different scenarios and have different capabilities.
```

Comment:
did we discuss whether to capitalize these? in 'use client' page, I didn't and I would lean towards not

Reply:
Dan suggested that Server Components and Client Components be capitalized but I don't have a strong opinion either way. I've changed this to be lower case for now.

Reply:
I'm finding it weird to be lower case. In my head, React Server Components is essentially a proper concept. It has a capitalized acronym; RSC. Similarly, SSR is Server Side Rendering, not server side rendering. 

When we've talked about RSC in past communication, like blog posts, it's been capitalized. https://react.dev/blog/2022/03/29/react-v18#server-components-is-still-in-development

> [Server Components](https://react.dev/blog/2020/12/21/data-fetching-with-react-server-components) is an upcoming feature that allows developers to build apps that span the server and client, combining the rich interactivity of client-side apps

https://react.dev/blog/2021/12/17/react-conf-2021-recap#streaming-server-rendering-with-suspense

> In the keynote, we also shared our vision for Suspense, Server Components, new React working groups, and our long-term many-platform vision for React Native.

Reply:
Haven’t caught up on all the messages here but I had been thinking “React Server Components” when talking about the feature but not when talking about the type of the component, if that makes sense.

Like if it was called Full-Stack React and had server components and client components. (Maybe that’s too confusing.)

Reply:
Yep, I agree! When talking about a specific server component it is lower case, when talking about React Server Components, it is upper case. I was thinking about it the same way.

When talking about a React feature being supported in a Server Component it is uppercase (but maybe that should always just be Server Component*s*)

Reply:
My inclination would be:

* Frameworks that support React Server Components let you write both server components and client components.
* Only server components can use async/await at present, but client component support is planned. [this is different [this is different from what you said @TheSavior]
* I wrote a client component today and it made me sad.

i.e. capitalize it only if and only if you could replace the usage with "The React Architecture" or similar (if that were an official term)

Reply:
I've changed it back to capitalize "Server Components" and "Client Components" to be consistent with the React blog:

* https://react.dev/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022#server-components
* https://react.dev/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023#react-server-components
* https://react.dev/blog/2022/03/29/react-v18#server-components-is-still-in-development
* https://react.dev/blog/2023/05/03/react-canaries#example-react-server-components

which has more history than the react.dev doc pages that reference lowercase client component or server component:

* https://react.dev/reference/react/use-server#caveats
* https://react.dev/reference/react/useId#why-is-useid-better-than-an-incrementing-counter
* https://react.dev/reference/react/use-client#caveats

Reply:
ok, sgtm – not convinced we should capitalize these long term (eg: we don't capitalize component, element, context) but seems fine for now, thanks for investigating